### PR TITLE
Codegen: subproject cleanups and chdir

### PR DESCRIPTION
**Pull request by thockin**
Three related commits, building on #116280

1) Move the hack/ directory and scripts within k8s.io/code-generator/examples to be more self-contained, and fix the fallout.

2) When running subproject generators, chdir and pretend it's a standalone repo (as an external user would use it)

3) Indent the subproject generation logs for easier reading.

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

**Comment by thockin** [comment]
Rebased

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -883,19 +883,22 @@ function codegen::informers() {
...
+        vendor/k8s.io/apiextensions-apiserver
+        vendor/k8s.io/metrics
+        vendor/k8s.io/apiextensions-apiserver/examples/client-go
     )
 
-    for s in "${subs[@]}"; do 
-        CODEGEN_PKG=./vendor/k8s.io/code-generator "$s"
+    for sub in "${subs[@]}"; do
+        local codegen
+        codegen="$(pwd)/vendor/k8s.io/code-generator"
```
Comment (aojea) [inline hack/update-codegen.sh:905]: why is this inside the loop? is not always the same value?

Comment (thockin) [reply hack/update-codegen.sh:905]: haha, yes. :)

**Review by thockin — COMMENTED** [review]
Will push a new update

**Pull request merged**